### PR TITLE
Add type hints to `test_authentication.py`

### PR DESCRIPTION
**Pull request by TechNiick**
# Summary

Related to [this](https://github.com/encode/starlette/pull/2445)
Type annotation added to `test_authentication.py`

# Checklist

- [x] I understand that this PR may be closed in case there was no previous discussion. (This doesn't apply to typos!)
- [ ] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [ ] I've updated the documentation accordingly.

### CODE REVIEW

File: tests/test_authentication.py [hunk 1]
```diff
@@ -1,9 +1,16 @@
 import base64
 import binascii
+import sys
+from typing import Any, Awaitable, Callable, Optional, Tuple
```
Comment (Kludex) [inline tests/test_authentication.py:3]: Let's avoid use `Any`, please.

File: tests/test_authentication.py [hunk 2]
```diff
@@ -15,14 +22,23 @@
...
 from starlette.middleware.authentication import AuthenticationMiddleware
-from starlette.requests import HTTPConnection
+from starlette.requests import HTTPConnection, Request
 from starlette.responses import JSONResponse
 from starlette.routing import Route, WebSocketRoute
-from starlette.websockets import WebSocketDisconnect
+from starlette.testclient import TestClient
+from starlette.websockets import WebSocket, WebSocketDisconnect
+
+P = ParamSpec("P")
```
Comment (Kludex) [inline tests/test_authentication.py]: Is it being used?

**Review by Kludex — CHANGES_REQUESTED** [review]

File: tests/test_authentication.py (hunk 2 above)
Comment (TechNiick) [reply tests/test_authentication.py]: Yes, I have added it on `**kwargs: P.kwargs`.

File: tests/test_authentication.py [hunk 3]
```diff
@@ -152,9 +172,9 @@ def decorated_sync(request, additional):
     )
 
 
-def ws_inject_decorator(**kwargs):
-    def wrapper(endpoint):
-        def app(websocket):
+def ws_inject_decorator(**kwargs: P.kwargs) -> Callable[..., AsyncEndpoint]:  # type: ignore
```
Comment (Kludex) [inline tests/test_authentication.py]:
```suggestion
def ws_inject_decorator(**kwargs: Any) -> Callable[..., AsyncEndpoint]:
```

Comment (Kludex) [reply tests/test_authentication.py]: P is wrong.

**Review by Kludex — APPROVED** [review]

**Pull request merged**